src/MessagePack/MessagePackReader.cs

```diff
@@ -29,7 +29,7 @@ namespace MessagePack
         /// <param name="memory">The buffer to read from.</param>
         public MessagePackReader(ReadOnlyMemory<byte> memory)
         {
-            this.reader = new SequenceReader<byte>(memory);
+            this.reader = new SequenceReader<byte>(new ReadOnlySequence<byte>(memory));
```

Comment:
`SequenceReader` that ships in netcoreapp3.0 does not have an overload that accepts a `ReadOnlyMemory<T>`: https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Buffers/SequenceReader.cs. This avoids having to ifdef when targeting a newer TFM.

Reply:
I recently added that ctor overload because `ReadOnlySequence<T>.ctor` is actually quite slow (relative to a fast deserialization operation). Our `SequenceReader<T>` is a superset of what's in corefx, so we'll keep that rather than use .NET Core 3.0's version. No `#ifdef` should be necessary. We just need to explicitly use our copy if there's a simple type name collision.

Reply:
See #57 for the change where this overload was added, and the data for the significant perf improvement I refer to (41% faster for small deserialization jobs!)

Reply:
>  We just need to explicitly use our copy if there's a simple type name collision.

Let me see what that looks like. In general, I was attempting to avoid having code that we don't absolutely need for the specific scenarios I'm attempting to address.

Reply:
@AArnott lets file an API addition to SequenceReader /cc @JeremyKuhne

Reply:
@StephenToub \ @davidfowl could the change to SequenceReader in #57 go to CoreFx? Should I file an issue for this?

Reply:
Always file an issue, infact, file an API request 😄

Reply:
Done - https://github.com/dotnet/corefx/issues/36423.